server/mock/datastore_mock.go

```diff
@@ -3122,6 +3127,11 @@ func (s *DataStore) DeleteMDMAppleConfigProfile(ctx context.Context, profileID u
 	return s.DeleteMDMAppleConfigProfileFunc(ctx, profileID)
 }
 
+func (s *DataStore) GetHostMDMProfiles(ctx context.Context, hostUUID string) ([]fleet.HostMDMAppleProfile, error) {
+	s.GetHostMDMProfilesFuncInvoked = true
+	return s.GetHostMDMProfilesFunc(ctx, hostUUID)
```

Comment:
I think you need to use `make generate-mock` or update your local mock version (check the makefile to grab the commit)

Newer mocks include a lock here